### ISSUE TITLE
Invalid universal2 wheel

### ISSUE THREAD

**Issue by ermakov-oleg** [Stranger6667/jsonschema-rs#409]
Wheel `jsonschema_rs-0.16.2-cp311-cp311-macosx_11_0_universal2.whl` not universal.

```
 ImportError: dlopen(/<path>/python3.11/site-packages/jsonschema_rs/_jsonschema_rs.cpython-311-darwin.so, 0x0002): tried: '/<path>/python3.11/site-packages/jsonschema_rs/_jsonschema_rs.cpython-311-darwin.so' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))
```

**Comment by Stranger6667** [comment]
Thanks for reporting!

I was thinking about overhauling of how Python bindings are packaged, particularly, moving to `maturin` and using something like `cibuildwheels`. However, I also wanted to switch to `abi3` to decrease the number of wheels, but `PyO3` doesn't support extending base types with the `abi3` feature yet (needed for `ValidationError`), so I put it aside for now.

But, I think we just need to migrate to `maturin` and fix wheels without switching to the `abi3` feature. Or, if it is feasible - fix macOs wheels in the current approach (I guess it requires macOs M1 runner or QEMU)

**Comment by Stranger6667** [comment]
Should be fixed in `0.16.3`. Feel free to reopen this issue if it would not work for you.
